zrml/prediction-markets/src/lib.rs

```diff
@@ -2390,11 +2461,56 @@ mod pallet {
             // If the oracle reported right, return the OracleBond, otherwise slash it to
             // pay the correct reporters.
             let mut overall_imbalance = NegativeImbalanceOf::<T>::zero();
-            if report.by == market.oracle && report.outcome == resolved_outcome {
-                Self::unreserve_oracle_bond(market_id)?;
+
+            let report_by_oracle = report.by == market.oracle;
+            let is_correct = report.outcome == resolved_outcome;
+
+            // TODO: this wrapper (if let Some and !settled) can be removed
+            // TODO: if there are only markets with outsiders, who all reserved OutsiderBond
+            let outsider_bond_pending = || -> bool {
+                if let Some(bond) = &market.bonds.outsider {
+                    if !bond.is_settled {
+                        return true;
+                    }
+                }
+                false
+            };
+
+            let unreserve_outsider = || -> DispatchResult {
+                if outsider_bond_pending() {
```

Comment:
Should we warn if the outsider bond is not pending? Seems like a serious logic error to me.

Reply:
9e42f0d25f51ad1c975798dc3e44b7915d91d367

Reply:
If we warn, then this warning will come for every market, that has an outsider reporter (with no bond). Are you okay with that? I think it depends on the amount of markets which are in the `Disputed` or `Reported` state, which have got outsiders. If the number is zero, then we are good to go. I just want to avoid the spam of these warnings for many resolution. But I expect less markets, which already have got outsider reports.

Reply:
The number of marktes should be very low, so I'm perfectly okay with keeping the warning.